lib/CrashHandler/CrashHandler.cpp

```diff
@@ -142,156 +145,84 @@ std::atomic<char const*> stateString = nullptr;
-/// Assumes that the buffer pointed to by s has enough space to
-/// hold the thread id, the thread name and the signal name
-/// (4096 bytes should be more than enough).
-size_t buildLogMessage(char* s, std::string_view context, int signal,
-                       siginfo_t const* info, void* ucontext) {
+void buildLogMessage(SmallString& buffer, std::string_view context, int signal,
+                     siginfo_t const* info, void* ucontext) {
   // build a crash message
-  char* p = s;
-  appendNullTerminatedString("💥 ArangoDB ", p);
-  appendNullTerminatedString(ARANGODB_VERSION_FULL, p);
-  appendNullTerminatedString(", thread ", p);
-  p += arangodb::basics::StringUtils::itoa(
-      uint64_t(arangodb::Thread::currentThreadNumber()), p);
+  buffer.append("💥 ArangoDB ").append(ARANGODB_VERSION_FULL);
+  buffer.append(", thread ")
+      .appendUInt64(uint64_t(arangodb::Thread::currentThreadNumber()));
 
 #ifdef __linux__
   arangodb::ThreadNameFetcher nameFetcher;
-  std::string_view name = nameFetcher.get();
-  appendNullTerminatedString(" [", p);
-  appendNullTerminatedString(name, p);
-  appendNullTerminatedString("]", p);
+  buffer.append(" [").append(nameFetcher.get()).append("]");
 #endif
 
   bool printed = false;
-  appendNullTerminatedString(" caught unexpected signal ", p);
-  p += arangodb::basics::StringUtils::itoa(uint64_t(signal), p);
-  appendNullTerminatedString(" (", p);
-  appendNullTerminatedString(arangodb::signals::name(signal), p);
+  buffer.append(" caught unexpected signal ").appendUInt64(uint64_t(signal));
+  buffer.append(" (").append(arangodb::signals::name(signal));
 #ifndef _WIN32
   if (info != nullptr) {
-    appendNullTerminatedString(") from pid ", p);
-    p += arangodb::basics::StringUtils::itoa(uint64_t(info->si_pid), p);
+    buffer.append(") from pid ").appendUInt64(uint64_t(info->si_pid));
     printed = true;
   }
 #endif
   if (!printed) {
-    appendNullTerminatedString(")", p);
+    buffer.append(")");
   }
 
-  char const* ss = ::stateString.load();
-  if (ss != nullptr) {
-    appendNullTerminatedString(" in state \"", p);
-    appendNullTerminatedString(ss, p);
-    appendNullTerminatedString("\" ", p);
+  if (char const* ss = ::stateString.load(); ss != nullptr) {
+    buffer.append(" in state \"").append(ss).append("\" ");
   }
 
 #ifndef _WIN32
   if (info != nullptr && (signal == SIGSEGV || signal == SIGBUS)) {
     // dump address that was accessed when the failure occurred (this is
     // somewhat likely a nullptr)
-    appendNullTerminatedString(" accessing address 0x", p);
-    unsigned char const* x =
-        reinterpret_cast<unsigned char const*>(info->si_addr);
-    unsigned char const* s = reinterpret_cast<unsigned char const*>(&x);
-    appendHexValue(s, sizeof(unsigned char const*), p, false);
+    buffer.append(" accessing address 0x");
+    auto const* x = reinterpret_cast<void const*>(info->si_addr);
+    buffer.appendHexValue(x, false);
```

Comment:
Wouldn't this suffice?
```suggestion
    buffer.appendHexValue(info->si_addr, false);
```